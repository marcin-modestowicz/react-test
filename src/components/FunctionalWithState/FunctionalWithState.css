.container {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}

h2 {
  margin: 2rem;
}
