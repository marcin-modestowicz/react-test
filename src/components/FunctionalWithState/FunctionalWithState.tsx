import React, { useState } from 'react';
import './FunctionalWithState.css';

const FunctionalWithState = () => {
  const [email, handleEmailChange] = useState('');
  const [password, handlePasswordChange] = useState('');

  return (
    <div className="container">
      <form>
        <h2>FunctionalWithState</h2>
        <div>
          <input
            id="email"
            value={email}
            onChange={e => handleEmailChange(e.currentTarget.value)}
          />
          <label htmlFor="email">Email</label>
        </div>
        <div>
          <input
            id="password"
            type="password"
            value={password}
            onChange={e => handlePasswordChange(e.currentTarget.value)}
          />
          <label htmlFor="password">Password</label>
        </div>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
};

export default FunctionalWithState;
