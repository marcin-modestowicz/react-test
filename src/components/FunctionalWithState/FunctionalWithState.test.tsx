import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FunctionalWithState from "./FunctionalWithState";

describe("FunctionalWithState component", () => {
  test("should render", () => {
    const functionalWithState = shallow(
      <FunctionalWithState/>
    );
    expect(toJson(functionalWithState)).toMatchSnapshot();
  });

  test("should render with data", () => {
    const functionalWithState = shallow(
      <FunctionalWithState/>
    );

    functionalWithState.find('#email').simulate('change', { currentTarget: { value: 'email' }});
    functionalWithState.find('#password').simulate('change', { currentTarget: { value: 'password' }});

    expect(toJson(functionalWithState)).toMatchSnapshot();
  });
});
