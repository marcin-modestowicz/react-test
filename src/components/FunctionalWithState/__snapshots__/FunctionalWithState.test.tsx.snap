// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalWithState component should render 1`] = `
<div
  className="container"
>
  <form>
    <h2>
      FunctionalWithState
    </h2>
    <div>
      <input
        id="email"
        onChange={[Function]}
        value=""
      />
      <label
        htmlFor="email"
      >
        Email
      </label>
    </div>
    <div>
      <input
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`FunctionalWithState component should render with data 1`] = `
<div
  className="container"
>
  <form>
    <h2>
      FunctionalWithState
    </h2>
    <div>
      <input
        id="email"
        onChange={[Function]}
        value="email"
      />
      <label
        htmlFor="email"
      >
        Email
      </label>
    </div>
    <div>
      <input
        id="password"
        onChange={[Function]}
        type="password"
        value="password"
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
